src/ZoneDifficulty.h

```diff
@@ -3,40 +3,134 @@
 
 #include "Player.h"
 #include "Config.h"
+#include "InstanceScript.h"
 #include "ScriptMgr.h"
 #include "ScriptedGossip.h"
 
-struct ZoneDifficultyData
+struct ZoneDifficultyNerfData
 {
     float HealingNerfPct;
     float AbsorbNerfPct;
     float SpellDamageBuffPct;
     float MeleeDamageBuffPct;
-    bool Enabled;
+    int8 Enabled;
+    float HealingNerfPctHard;
+    float AbsorbNerfPctHard;
+    float SpellDamageBuffPctHard;
+    float MeleeDamageBuffPctHard;
+};
+
+struct ZoneDifficultyHardmodeInstData
+{
+    bool HardmodeOn;
+    bool HardmodePossible;
+};
+
+struct ZoneDifficultyHardmodeMapData
+{
+    uint32 EncounterEntry;
+    uint32 OverrideGO;
+    uint32 RewardType;
+};
+
+struct ZoneDifficultyRewardData
+{
+    uint32 Entry;
+    uint32 Price;
+    uint32 Enchant;
+    uint8 EnchantSlot;
+    uint32 Achievement;
 };
 
 const int32 DUEL_INDEX = 0x7FFFFFFF;
 const int32 DUEL_AREA = 2402;       // Forbidding Sea (Wetlands)
 
+const uint32 NPC_TEXT_LEADER_NORMAL = 91301;
```

Comment:
depending on how nitty gritty you want to be, according to AC C++ standards

> const keyword should always go after type name

https://www.azerothcore.org/wiki/cpp-code-standards#defining-const-variables

so `uint32 const variable`